.github/workflows/build.yml

```
@@ -0,0 +1,22 @@
+name: build
+
+on:
+  push:
+    branches: 
+      - main
+  pull_request:
+    branches:
+      - "*"
```

```suggestion
on: [push, pull_request]
```

this would be simpler and would run builds on all branches - not only for pull requests targeting `main`

---

I don't want to build every branch. I frequently push changes which cannot be built. I push them only to not lose them and to make them accessible from multiple computers. When the change is "ready" to build or somebody would like to check changes - a PR should be created (a normal one or draft). This strategy prevents wasting resources on unneded builds 🙂

---

```suggestion
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
```

probably this will result in the same and with less typing 😉

---

I've used it.